docs/src/main/asciidoc/secretmanager.adoc

```diff
@@ -0,0 +1,83 @@
+
+NOTE: To begin using this feature, you must set the setting `spring.cloud.gcp.secretmanager.bootstrap.enabled` to **true**, this feature disabled by default.
+
+Spring Cloud GCP will load the **latest** version of each secret into the application context.
+
+All secrets will be loaded into the application environment using their `secretId` as the property name.
+For example, if your secret's id is `my-secret` then it will be accessible as a property using the name `my-secret`.
+If you would like to append a prefix string to all property names imported from Secret Manager, you may use the `spring.cloud.gcp.secretmanager.secret-name-prefix` setting described below.
+
+Spring Cloud GCP Secret Manager offers several configuration properties to customize the behavior.
+
+|===
+| Name | Description | Required | Default value
+| `spring.cloud.gcp.secretmanager.bootstrap.enabled` | Enables loading secrets from Secret Manager as a bootstrap property source. Set this to **true** to enable the feature. | No | `false`
+| `spring.cloud.gcp.secretmanager.credentials.location` | OAuth2 credentials for authenticating with the Google Cloud Secret Manager API. | No | By default, infers credentials from https://cloud.google.com/docs/authentication/production[Application Default Credentials].
+| `spring.cloud.gcp.secretmanager.credentials.encoded-key` | Base64-encoded contents of OAuth2 account private key for authenticating with the Google Cloud Storage API. | No | By default, infers credentials from https://cloud.google.com/docs/authentication/production[Application Default Credentials].
+| `spring.cloud.gcp.secretmanager.project-id` | The GCP Project used to access Secret Manager API. | No | By default, infers the project from https://cloud.google.com/docs/authentication/production[Application Default Credentials].
+| `spring.cloud.gcp.secretmanager.secret-name-prefix` | A prefix string to prepend to the property names of secrets read from Secret Manager | No | "" (empty string)
+|===
+
+=== Secret Manager Template
+
+The `SecretManagerTemplate` class simplifies operations of creating, updating, and reading secrets.
+
+To begin using this class, you may autowire an instance of the class using `@Autowired` after adding the starter dependency to your project.
+
+[source, java]
+----
+@Autowired
+private SecretManagerTemplate secretManagerTemplate;
+----
+
+Please consult https://github.com/spring-cloud/spring-cloud-gcp/blob/master/spring-cloud-gcp-secretmanager/src/main/java/org/springframework/cloud/gcp/secretmanager/SecretManagerOperations.java[`SecretManagerOperations`] for information on what operations are available for the Secret Manager template.
+
+|===
```

Comment:
Is this section a duplicate?

Reply:
Parts of it were duplicated; I reorganized the properties so the duplicated authentication-related properties now appear in 1 place at the bottom.